src/test/fio/fio_ceph_objectstore.cc

```diff
@@ -482,6 +614,45 @@ Job::~Job()
   engine->deref();
 }
 
+void Job::check_throttle()
+{
+  if (subjob_number != 0)
+    return;
+
+  std::lock_guard<std::mutex> l(throttle_lock);
+  if (throttle_values.empty() && deferred_throttle_values.empty())
+    return;
+
+  if (ceph::mono_clock::is_zero(last) ||
+      ((cycle_throttle_period != cycle_throttle_period.zero()) &&
+       (ceph::mono_clock::now() - last) > cycle_throttle_period)) {
+    unsigned tvals = throttle_values.size() ? throttle_values.size() : 1;
+    unsigned dtvals = deferred_throttle_values.size() ? deferred_throttle_values.size() : 1;
+    unsigned total = tvals * dtvals;
+    if (!throttle_values.empty()) {
+      std::string val = std::to_string(throttle_values[index % tvals]);
+      std::cerr << "Setting bluestore_throttle_bytes to " << val << std::endl;
+      int r = engine->cct->_conf.set_val(
+	"bluestore_throttle_bytes",
+	val,
+	nullptr);
```

Comment:
How does value "bluestore_throttle_bytes" propagate from client (FIO) to server (osd/bluestore)?

Reply:
In this instance, the bluestore instance is in the same instance and this backend interacts with it directly.  There's no osd or objecter.  Modifying cct there directly updates bluestore.

Reply:
This doesn't seem problematic to me.  bluestore_throttle_bytes and _deferred_bytes are the same throttles the osd uses to regulate how much work bluestore will accept before blocking (indeed, the purpose here is essentially to determine what those throttles should be set to).  As such, there's already support for modifying them on the fly via injectargs etc.  Adding an additional interface seems superfluous.

Reply:
Also, there's no admin socket here since there's no osd.  The admin socket is created by the osd and simply uses this config modification mechanism.